dash/include/dash/Atomic.h

```diff
@@ -34,58 +39,56 @@ class Atomic
    * Default constructor.
    * Sets underlying global pointer to \c DART_GPTR_NULL.
    */
-  Atomic()
-  : _gptr(DART_GPTR_NULL),
-    _team(nullptr)
-  { }
+  AtomicAddress() = default;
 
   /**
-   * Constructor, creates a new instance of \c dash::Atomic from a DART
+   * Constructor, creates a new instance of \c dash::AtomicAddress from a DART
    * pointer.
    */
-  Atomic(
-    dart_gptr_t  gptr,
+  AtomicAddress(
+    const dart_gptr_t & gptr,
     dash::Team & team = dash::Team::All())
   : _gptr(gptr),
-    _team(&team)
+    _dart_teamid(team.dart_id())
   { }
 
   /**
-   * Constructor, creates a new instance of \c dash::Atomic from any object
+   * Constructor, creates a new instance of \c dash::AtomicAddress from any object
    * that is convertible to \c dart_gptr_t.
    */
   template<class GlobalType>
-  Atomic(
+  AtomicAddress(
     const GlobalType & global,
     dash::Team       & team   = dash::Team::All())
-  : Atomic(global.dart_gptr(), team)
+  : AtomicAddress(global.dart_gptr(), team)
   { }
 
-  Atomic(const self_t & other)           = default;
+  AtomicAddress(const self_t & other)    = default;
 
   self_t & operator=(const self_t & rhs) = default;
 
   /**
    * Set the value of the shared atomic variable.
    */
-  void set(ValueType val)
+  void set(ValueType value)
   {
-    DASH_LOG_DEBUG_VAR("Atomic.set()", val);
-    DASH_LOG_TRACE_VAR("Atomic.set",   _gptr);
-    *(reference(_gptr)) = val;
-    DASH_LOG_DEBUG("Atomic.set >");
+    DASH_LOG_DEBUG_VAR("AtomicAddress.set()", value);
+    DASH_LOG_TRACE_VAR("AtomicAddress.set",   _gptr);
+    *(pointer(_gptr)) = value;
+    dart_flush_all(_gptr);
+    DASH_LOG_DEBUG("AtomicAddress.set >");
   }
 
   /**
    * Get a reference on the shared atomic value.
    */
   reference get()
   {
-    DASH_LOG_DEBUG("Atomic.get()");
-    DASH_LOG_TRACE_VAR("Atomic.get", _gptr);
+    DASH_LOG_DEBUG("AtomicAddress.get()");
+    DASH_LOG_TRACE_VAR("AtomicAddress.get", _gptr);
     DASH_ASSERT(!DART_GPTR_ISNULL(_gptr));
     reference ref(_gptr);
```

Comment:
See my comment on `set()`. Better use `dart_fetch_and_op` with `DART_OP_NONE`. Alternatively, we might consider implementing a counterpart of `MPI_Get_accumulate`, i.e., `dart_get_accumulate`.